python/src/nanoarrow/iterator.py

```diff
@@ -244,6 +244,126 @@ def _binary_iter(self, offset, length):
             for start, end in zip(starts, ends):
                 yield bytes(data[start:end])
 
+    def _date_iter(self, offset, length):
+        from datetime import date, timedelta
+
+        storage = self._primitive_iter(offset, length)
+        epoch = date(1970, 1, 1)
+
+        if self._schema_view.type_id == CArrowType.DATE32:
+            for item in storage:
+                if item is None:
+                    yield item
+                else:
+                    yield epoch + timedelta(item)
```

Comment:
`datetime.date` also has a `fromtimestamp` method that accepts seconds since epoch, so you could also use that, which seems to be slightly faster:

```
In [53]: item = 10957

In [54]: epoch = datetime.date(1970, 1, 1)

In [55]: epoch + datetime.timedelta(item)
Out[55]: datetime.date(2000, 1, 1)

In [56]: datetime.date.fromtimestamp(60*60*24*item)
Out[56]: datetime.date(2000, 1, 1)

In [57]: %timeit epoch + datetime.timedelta(item)
164 ns ± 0.569 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [58]: %timeit datetime.date.fromtimestamp(60*60*24*item)
118 ns ± 0.0869 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```

Reply:
And the same for DATE64 if converting the milliseconds to seconds

Reply:
Hmm, good point. But this is true for `datetime.datetime.fromtimestamp` as well, though ..